src/sbd-inquisitor.c

```diff
@@ -1044,12 +1044,12 @@ int main(int argc, char **argv, char **envp)
 		case 'w':
                         cl_log(LOG_NOTICE, "Using watchdog device '%s'", watchdogdev);
                         free(watchdogdev);
-                        watchdogdev = strdup(optarg);
+                        watchdogdev = strdup(sanitize_option_value(optarg));
```

Comment:
atoi checks for NULL. But actually we wouldn't need stripping away space in front with it as it does that as well. Anyway I guess using sanitation for ints makes sense as we might improve the sanitation with whatever in future.
But unfortunately strdup doesn't check for NULL afaik while getopt iirc provides that optarg is !=NULL.
Wherever environment sanitation was used before it checks for NULL.
But anyway silently setting to 0 with an empty string wasn't that pretty ;-)
Maybe we could functionalize parsing an int using strtol including checking *endtr == '\0' to detect garbage. Guess we don't have that so far in sbd and pacemaker deprecated all the convenience function stuff.

Reply:
> atoi checks for NULL. But actually we wouldn't need stripping away space in front with it as it does that as well. Anyway I guess using sanitation for ints makes sense as we might improve the sanitation with whatever in future.

Was the idea :-)

> But unfortunately strdup doesn't check for NULL afaik while getopt iirc provides that optarg is !=NULL.
> Wherever environment sanitation was used before it checks for NULL.

Indeed, the purpose is to treat a setting with an empty string as being defaulted. We probably should do the same for command line options.

> But anyway silently setting to 0 with an empty string wasn't that pretty ;-)
> Maybe we could functionalize parsing an int using strtol including checking *endtr == '\0' to detect garbage. Guess we don't have that so far in sbd and pacemaker deprecated all the convenience function stuff.

Probably let's do NULL checks for now to at least avoid overriding the sensible defaults.

Reply:
Not so sure if command-line parameter without argument treated as default is intuitive.
Haven't seen that behavior at least elsewhere.